resources/git-fork.md

```diff
@@ -0,0 +1,98 @@
+# Git: Forking a repository
+
+> _Ah, the life! Branching, committing, pushing, pulling, merging... modifying code to my hearts' content... breathing life into features like some sort of magician or squashing bugs like an exterminator. Bliss. Until..._
+
+```bash
+! [remote rejected] master -> master (protected branch hook declined)
+error: failed to push some refs to 'git@github.com:some-account/some-repo.git'
+```
+
+☹️
+
+When working with code, there are times when you will want to modify code that resides in a repository that is not under your control, or with which for a variety of reasons you don't wish to interact. You might want to introduce features that are outside the scope of the project. You may want to build something completely unrelated and just use the repository as a starting point. Maybe a useful library needs a bug fix, but the maintainer has abandoned the repository. 
+
+For all these times and [more](#why-fork), there's **Forking**!
+
+## What is a Fork?
+
+Forking is creating a new copy of the entire repository. This might sound like `git clone`, and behind the scenes it is. The difference is in where the new copy resides. A "clone" typically refers to a _local_ copy of some repository. A _fork_, in contrast, is a second *remote* repository distinct from the original that it copies.
+
+Historically, a _fork_ was created when a developer or team wanted to take a project in a new direction, or carry on an original mission where the core project undergoes some major shift. Some examples of this are NodeJS and IO.js, or when the licensing changes to `OpenOffice` by Sun led to the LibreOffice fork, and ultimately OpenOffice's demise. These classic situations can be referred to as **hard forks**, where the forked code is not meant to be integrated back into the original (upstream) codebase. What we deal with most often are so called **soft forks**, where we _do_ intend for our changes to be integrated back into the original repository, and the fork is a convenient system for creating an isolated workspace. From here on when we say "fork" we will be talking about soft forks, unless explicitly stated otherwise.
```

Comment:
Maybe ` carry on an original mission where the core project` should be ` carry on an original mission while the core project` ?

Reply:
I like the hard fork examples but feel like some context is missing, can we link out to some other material or elaborate briefly on the relation between NodeJS and IO.js for example (if you've never heard of IO.js, you won't understand what this is saying?)

Reply:
Good point. Reworded (and corrected!) plus added links for context.